### PR TITLE
feat: provide authz model

### PR DESCRIPTION
**Pull request by danielvladco**

### CODE REVIEW

File: backend/pkg/modules/internal/bfla/init.go [hunk 1]
```diff
@@ -303,6 +303,36 @@ func (h httpHandler) GetEvent(w http.ResponseWriter, r *http.Request, eventID in
...
+	defer r.Body.Close()
+
+	ctx := r.Context()
+	select {
+	case <-ctx.Done():
+		httpResponse(w, http.StatusCreated, &restapi.ApiResponse{Message: fmt.Sprintf("the request took too long: %s", ctx.Err())})
+	default:
+		apiinfo, err := h.accessor.GetAPIInfo(r.Context(), uint(apiID))
+		if err != nil {
+			log.Error("error getting openAPI spec")
```
[the remark this answers is not in the document]
Comment (FrimIdan) [reply backend/pkg/modules/internal/bfla/init.go:317]: IMO it's not best practice to count on another function behaviour.

[the remark this answers is not in the document]
Comment (danielvladco) [reply backend/pkg/modules/internal/bfla/init.go:317]: done

File: backend/pkg/modules/internal/bfla/init.go [hunk 2]
```diff
@@ -303,6 +303,37 @@ func (h httpHandler) GetEvent(w http.ResponseWriter, r *http.Request, eventID in
...
+		} else {
+			specType = bfladetector.SpecTypeFromAPIInfo(apiinfo)
+		}
+		if specType == bfladetector.SpecTypeNone {
+			httpResponse(w, http.StatusOK, &restapi.ApiResponse{Message: "Spec not found, please either provide or reconstruct an api spec"})
+			return
+		}
+		authModelReq := &restapi.AuthorizationModel{}
+		if err := json.NewDecoder(r.Body).Decode(authModelReq); err != nil {
+			httpResponse(w, http.StatusNotFound, &restapi.ApiResponse{Message: fmt.Sprintf("error decoding body; id=%d err: %s", apiID, err)})
```
Comment (klyr) [inline backend/pkg/modules/internal/bfla/init.go:327]:
```suggestion
			httpResponse(w, http.StatusBadRequest, &restapi.ApiResponse{Message: fmt.Sprintf("error decoding body; id=%d err: %s", apiID, err)})
```

**Review by klyr — APPROVED** [review]
LGTM

**Pull request closed without merging**